src/sdi-refresh-monitor.c

```diff
@@ -224,6 +252,161 @@ static void refresh_change(gpointer p) {
                                 g_object_ref(data->self));
 }
 
+static gboolean status_is_done(const gchar *status) {
+  gboolean done = g_str_equal(status, "Done") | g_str_equal(status, "Abort") |
+                  g_str_equal(status, "Error") | g_str_equal(status, "Hold") |
+                  g_str_equal(status, "Wait");
+  return done;
+}
+
+static void update_inhibited_snaps(SdiRefreshMonitor *self, SnapdChange *change,
+                                   gboolean done, gboolean hold) {
+  SnapdAutorefreshChangeData *changeData =
+      SNAPD_AUTOREFRESH_CHANGE_DATA(snapd_change_get_data(change));
+
+  if (changeData == NULL) {
+    return;
+  }
+
+  GStrv snapNames = snapd_autorefresh_change_data_get_snap_names(changeData);
+  for (gchar **p = snapNames; *p != NULL; p++) {
+    g_autoptr(SdiSnap) snap = find_snap(self, *p);
+    // Only show progress bar if that snap shown an 'inhibited' notification
+    if (snap == NULL)
+      continue;
+
+    if (!sdi_snap_get_inhibited(snap))
+      continue;
+
+    if (done || hold) {
+      remove_snap(self, snap);
+      SnapRefreshData *data = g_malloc0(sizeof(SnapRefreshData));
```

Comment:
Would be a little cleaner with a `snap_refresh_data_new()`.

Reply:
You are right. Done.